clause/label/label.go

```diff
@@ -16,14 +20,14 @@ var (
 		Summary: "Searches based on an object's label (typically, its filename)",
 		Args: map[string]clause.ClauseArgumentDocumentation{
 			"label": clause.ClauseArgumentDocumentation{Type: "string", Summary: "The label to search for"},
+			"exact": clause.ClauseArgumentDocumentation{Type: "boolean", Summary: "Whether to search more precisely, or whether the query should be processed to add wildcards"},
 		},
 	}
 )
 
 type LabelArgs struct {
```

Comment:
The current mapping for the `label` field is shown below.
In the future, we could potentially add new subfields or add modify analyzers.  
I'm not suggesting any changes now, only that the potential for mapping changes be considered.
Worst case, mappings change and you write a new clause, or not.
Food for thought.

----

```
{
  "data-archer-2017.05.21": {
    "mappings": {
      "folder": {
        ...
        "label": {
          "type": "text",
          "analyzer": "irods_entity"
        }
        ...
      },
      "file": {
        ...
        "label": {
          "type": "text",
          "analyzer": "irods_entity"
        }
        ...
      }
    ...
    },
    "settings": {
      "index": {
        "analysis": {
          "analyzer": {
            ...
            "irods_entity": {
              "filter": [
                "asciifolding",
                "lowercase"
              ],
              "type": "custom",
              "tokenizer": "irods_entity"
            }
          }
        }
      }
    }
  }
}
```

Reply:
Yeah; for now I figured I'd mostly copy what the UI query builder produces. I could see adding a new clause which uses the 'label' field (or a subfield that tokenizes/analyses differently, more likely) if we wanted to add search by file extension, or such. I could also see it being useful for the "exact" search, if we want case-sensitivity at some point (that's a new feature, so making it more exact over time seems like it could be good).